ibis/common/patterns.py

```diff
@@ -275,6 +294,160 @@ def __rich_repr__(self):
             yield name, getattr(self, name)
 
 
+class Creator(Hashable):
+    __slots__ = ()
+
+    @abstractmethod
+    def __eq__(self, other):
+        ...
+
+    @abstractmethod
+    def make(self, context):
```

Comment:
This looks like it is the primary interface for this class, so it should have a docstring.

Reply:
Going to add. If you have a better name suggestion for the `Creator` base class then please come up with a good verb for this method too.